src/edu/csus/ecs/pc2/core/PacketHandler.java

```diff
@@ -3536,7 +3536,26 @@ private void checkoutRun(Packet packet, Run run, ClientId whoRequestsRunId, bool
                     // just get run and sent it to them.
 
                     theRun = contest.getRun(run.getElementId());
-                    RunFiles runFiles = contest.getRunFiles(run);
+                    
+                    RunFiles runFiles = null;
+                    
+                    // in case GetRunFiles throws an exception we want to deal with it separately
+                    try {
+                        runFiles = contest.getRunFiles(run);  
+                    } catch (Exception e) {
+                        controller.getLog().info("contest.getRunFiles (R/O) can not get files for run " + run.getNumber() + ": " + e.getMessage());
+                        
+                        try {
+                            contest.cancelRunCheckOut(run, whoRequestsRunId);
```

Comment:
For the "non-readOnly" case (below), after calling `cancelRunCheckOut()` the code sets the run Status to `NEW` (and sends an "available NEW run" packet to judgedAndOthers.  This was what I think we all agreed should happen (for the non-readOnly case.  Is there a reason for NOT doing that here?